src/main/java/duke/parsers/Parser.java

```diff
@@ -0,0 +1,154 @@
+     * @throws EmptyTaskDoneException If the user inputs done without a task.
+     * @throws EmptyTaskDeletedException If the user inputs deleted without a task.
+     * @throws EmptyDueDateException If the user inputs deadline without a due date.
+     * @throws EmptyEventDateException If the user inputs an event without the date and/or time.
+     * @throws EmptySearchWordException If the user inputs find without a search word.
+     */
+    public static Command parse(String userInput) throws EmptyTaskDescriptionException, DukeDateTimeParseException,
+            EmptyTaskDoneException, EmptyTaskDeletedException,
+            EmptyDueDateException, EmptyEventDateException, EmptySearchWordException {
+
+        String[] arr = userInput.strip().split("\\s+", 2);
+        switch (arr[0].strip().toLowerCase()) {
+        case "bye":
+            return parseBye();
+        case "list":
+            return parseList();
+        case "done":
+            if (arr.length < 2) {
+                throw new EmptyTaskDoneException();
+            }
+            return parseDone(arr[1].strip());
+        case "todo":
+        case "deadline":
+        case "event":
+            if (arr.length < 2) {
+                throw new EmptyTaskDescriptionException(arr[0].strip());
+            }
+            return parseAdd(arr[0].strip(), arr[1].strip());
+        case "delete":
+            if (arr.length < 2) {
+                throw new EmptyTaskDeletedException();
+            }
+            return parseDelete(arr[1].strip());
+        case "today":
+            return parseToday();
+        case "find":
+            if (arr.length < 2) {
+                throw new EmptySearchWordException();
+            }
+            return parseFind(arr[1].strip().toLowerCase());
+        default:
+            throw new InvalidCommandException();
+        }
+    }
+
+    private static Command parseToday() {
+        return new TodayCommand();
+    }
+
+    private static AddCommand parseAdd(String commandName, String arguments)
+            throws EmptyTaskDescriptionException, DukeDateTimeParseException,
+            EmptyEventDateException, EmptyDueDateException {
+        Task task = null;
+        if (arguments.equals("")) {
+            throw new EmptyTaskDescriptionException(commandName);
+        }
+        switch (commandName) {
+        case "todo":
+            task = new Todo(arguments);
+            break;
+        case "deadline": {
+            String[] parsed = arguments.split(" /by ");
+            if (arguments.startsWith("/by")) {
+                throw new EmptyTaskDescriptionException(commandName);
+            }
+            if (parsed.length < 2) {
+                throw new EmptyDueDateException();
+            }
+            task = new Deadline(parsed[0], DukeDateTimeParser.parse(parsed[1]));
+            break;
+        }
+        case "event": {
+            String[] parsed = arguments.split(" /at ");
```

Comment:
Maybe `parsedInputParts` is a better choice for this type of arrays since it needs to be plural.
Other `case` code blocks also have the same issue.